### ISSUE TITLE
Resources Times Warning

### ISSUE THREAD

**Issue by KingRyuShin** [ESX-Org/esx_skin#16]
I don't know what is the cause..
but esx_skin is giving a resource time warning that cause it to use 65Mb of memory

**Comment by Kirouw** [comment]
Yes, same for me, someone fix it ? It's just when I got the big menu when you create your character ! 

Thanks guys !

**Comment by schwim0341** [comment]
Happens across all machines and installs that I've been a part of for the last couple of months.  Relogging after character creation is required to get rid of it.

**Comment by SLuuuuuuG** [comment]
Anybody got a fix for the resource time warning ?   Been dealing with it for months.  Would love to resolve it once and for all.

**Comment by Carlos221977** [comment]
I think this problem has many servers, mine too and I do not know how to solve it.

**Comment by greenleafrp** [comment]
ya has anyone made any progress on this error? or the creator? https://gyazo.com/b3805612a9c97230c1cb178233a45b6a

**Comment by ElPumpo** [comment]
Sorry for being so late on the subject, I personally don't use esx_skin so fixing it hasn't been a priority at all. I've (mostly) tracked this memory issue down to the callback functions returned by `skinchanger:getData` and `skinchanger:getSkin`. Avoiding the functions created by the calbacks seem to work.

Commit https://github.com/ESX-Org/esx_skin/commit/9b720acb607d616a1e6f890644f65011ae62829e implements this.

**Comment by HalCroves** [comment]
Hi ElPumpo, 
I'm sorry, but your commit 9b720ac didn't work correctly.

- Last **esx_menu_default** : Yep
- Last **esx_skin** : Yep
- last **skinchanger** : Yep

When you have fun doing left-right in the menu, the memory gets racing and never goes down again.

![unknown](https://user-images.githubusercontent.com/16627463/54643494-51899900-4a97-11e9-9f62-9e54ec007e6f.png)

**Comment by ElPumpo** [comment]
It isnt perfect but In my testing much better, which also is why I haven't closed this issue

**Comment by SLuuuuuuG** [comment]
I will agree it is much better but still not perfect.    Thank you for taking the time to work on this when you can.. The entire community will be forever in your debt.

**Comment by qalle-fivem** [comment]
I Fixed this by a full remake of the script.

First i replaced the Component list with something like this:

```lua
Config.Components = {
	{ ["label"] = "Sex", ["subMenu"] = "sex" },
	{ ["label"] = "Face", ["subMenu"] = "face" },
}
```

And second i added subMenus like this:

```lua
Config.SubMenus = {
	["sex"] = {
	  	{ ["label"] = "Sex", ["name"] = 'sex', ["value"] = 0, ["min"] = 0 }
	},

	["face"] = {
	  	{ ["label"] = "Face Type", ["name"] = 'face', ["value"] = 0,  ["min"] = 0 }
	}
}
```

When i have this i open a menu with all the components.

In that menu when i press enter it will trigger this:

```lua
		local subMenu = data.current.subMenu

		OpenSubMenu(subMenu)
```

which will open the subMenu like this:

```lua
	local openSub = Config.SubMenus[subMenu]
    local maxValues = GetMaxVals()

    local elements = {}

    for i = 1, #openSub do
        local component = openSub[i]

        if component ~= nil then
        	if yourAppearance[component["name"]] ~= nil then
            	table.insert(elements, { ["label"] = component["label"], ["value"] = yourAppearance[component["name"]], ["min"] = component["min"], ["max"] = maxValues[component["name"]], ["component"] = component["name"], ["textureof"] = component["textureof"], ["type"] = "slider" })
            else
				table.insert(elements, { ["label"] = component["label"], ["value"] = 0, ["min"] = component["min"], ["max"] = maxValues[component["name"]], ["component"] = component["name"], ["textureof"] = component["textureof"], ["type"] = "slider" })
            end
        end
    end
```

Now mine won't go above 1.82mb and when it goes above that it goes down to 800kb again.

I don't know if this helps but ye...

**Comment by Jason1123** [comment]
Man I have been trying all of these tips but nothing wants to change the slow speeds when changing character. i tried to change stuff on esx_menu_defaul, esx_skin, and even skinchanger...
